app/controllers/attachment_masks_controller.rb

```diff
@@ -0,0 +1,38 @@
+##
+# Controller to process FoiAttachment objects before being served publicly by
+# applying masks and censor rules.
+#
+class AttachmentMasksController < ApplicationController
+  before_action :set_no_crawl_headers
+  before_action :find_attachment
+  before_action :ensure_attachment, :ensure_referer
+
+  def wait
+    file = @attachment.masked_file
+
+    if file.attached?
+      redirect_to params[:referer]
+
+    else
+      FoiAttachmentMaskJob.perform_later(@attachment)
+    end
+  end
+
+  private
+
+  def set_no_crawl_headers
+    headers['X-Robots-Tag'] = 'noindex'
+  end
+
+  def find_attachment
+    @attachment = GlobalID::Locator.locate_signed(params[:id])
+  end
+
+  def ensure_attachment
+    raise ActiveRecord::RecordNotFound unless @attachment
+  end
+
+  def ensure_referer
+    raise RouteNotFound unless params[:referer].present?
```

Comment:
Is this supposed to prevent direct access of the `/attachments/:global_id` URL? I can still access it by pasting it into a new private browsing window. Is that what's expected?